ros2_batch_job/packaging.py

```diff
@@ -112,9 +112,21 @@ def build_and_test_and_package(args, job):
     # Only on Linux and OSX Python scripts have a shebang line
     if args.os in ['linux', 'osx']:
         print('# BEGIN SUBSECTION: rewrite shebang lines')
+        filenames = []
+
         bin_path = os.path.join(args.installspace, 'bin')
-        for filename in os.listdir(bin_path):
-            path = os.path.join(bin_path, filename)
+        filenames.append([os.path.join(bin_path, filename) for filename in os.listdir(bin_path)])
```

Comment:
It might be good to add the same check as below `os.path.isfile` (just in case someone creates a subfolder under `bin`).